### DISCUSSION TITLE
Add ability to customize default operating system support for new/converted modules

### DISCUSSION THREAD

**Discussion by scotje** [puppetlabs/pdk#1003] [Feature Request]
Users should be able to customize what the default operating system support is for new and newly converted modules. Currently, the pre-selected options are hard-coded.

Futher, the solution should support customizing the operating system versions that are included in the generated `metadata.json`.

For example, currently if you select "RedHat based Linux" in the interview process, it adds the following entry to `metadata.json`:

```json
{
      "operatingsystem": "RedHat",
      "operatingsystemrelease": [
        "8"
      ]
}
```

It would be useful to be able to customize the contents of the `operatingsystemrelease` array. (For example, set it to `["5", "6", "7"]` or `["6", "7", "8"]`.

**Comment by natemccurdy** [comment]
Big +1 to this. The hard-coded list of operating systems in https://github.com/puppetlabs/pdk/blob/v2.1.0/lib/pdk/module/metadata.rb#L8 are not always relevant to PDK users.

And in some cases this list causes spec tests failures that are red-herrings due to the OS's not being the intended target of a module.
This is my main gripe with the hard-coded list... it's not customizable.

I work with teams that are 100% CentOS shops, and they build their internal Puppet modules to only support CentOS. We use a custom PDK module template, but when doing a `pdk new module`, we have to hand-edit the metadata.json to remove the non-CentOS OS's.

It would be better if we could specify a default list of OS's or even make the metadata.json _part of the PDK module template_.
